pkg/jobs/hive/hive.go

```diff
@@ -485,6 +748,91 @@ func validateUpgradeVersion(client clientv1.Client, clusterName string, curator
 	return nil
 }
 
+func waitForMCV(client clientv1.Client, clusterName string, clusterNamespace string, mcv *managedclusterviewv1beta1.ManagedClusterView, errMsg error) error {
+	for i := 1; i <= 5; i++ {
+		time.Sleep(utils.PauseFiveSeconds)
+		if err := client.Get(context.TODO(), types.NamespacedName{
+			Name:      clusterName,
+			Namespace: clusterNamespace,
+		}, mcv); err != nil {
+			if i == 5 {
+				klog.Warning(err)
+				return errMsg
+			}
+			klog.Warning(err)
+			continue
+		}
+		labels := mcv.ObjectMeta.GetLabels()
+		if len(labels) == 0 {
+			return errMsg
+		}
+		if _, ok := labels[MCVUpgradeLabel]; ok {
+			if mcv.Status.Result.Raw != nil {
+				break
+			}
+		} else {
+			return errMsg
+		}
+	}
+
+	return nil
+}
+
+func validateEUSUpgradeVersion(client clientv1.Client, clusterName string, curator *clustercuratorv1.ClusterCurator, isInterVersion bool) error {
+	if curator.Spec.Upgrade.DesiredUpdate == "" {
+		return errors.New(fmt.Sprintf("DesiredUpdate is required to run EUS to EUS upgrade for Curator %q", curator.Name))
+	}
+	desiredVersion, err := semver.Make(curator.Spec.Upgrade.DesiredUpdate)
+	if err != nil {
+		return err
+	}
+
+	intermediateVersion, err := semver.Make(curator.Spec.Upgrade.IntermediateUpdate)
+	if err != nil {
+		return err
+	}
+
+	managedClusterInfo := managedclusterinfov1beta1.ManagedClusterInfo{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: clusterName,
+		Name:      clusterName,
+	}, &managedClusterInfo); err != nil {
+		return err
+	}
+
+	klog.V(2).Info("kubevendor: ", managedClusterInfo.Status.KubeVendor)
+
+	if managedClusterInfo.Status.KubeVendor != "OpenShift" {
+		return errors.New("can not upgrade non openshift cluster")
+	}
+
+	currentVersion, err := semver.Make(managedClusterInfo.Status.DistributionInfo.OCP.Version)
```

Comment:
can you double check is this nilpointer safe?

Reply:
I tried checking for nil. It's not possible:
![image](https://github.com/stolostron/cluster-curator-controller/assets/38960034/2a5b6d7b-cc6e-43a3-90b2-3aac03c533fd)